src/spartan/batched_ppsnark.rs

```diff
@@ -216,10 +216,7 @@ where
   fn ck_floor() -> Box<dyn for<'a> Fn(&'a R1CSShape<E>) -> usize> {
     Box::new(|shape: &R1CSShape<E>| -> usize {
       // the commitment key should be large enough to commit to the R1CS matrices
-      std::cmp::max(
-        shape.A.len() + shape.B.len() + shape.C.len(),
-        std::cmp::max(shape.num_cons, 2 * shape.num_vars),
-      )
+      shape.A.len() + shape.B.len() + shape.C.len()
```

Comment:
I think there could be some degenerate circuits where `|A|+|B|+|C|` is smaller than `2*num_vars`, in which case the pp_snark would commit to vectors larger than the commitment key support.

Reply:
Fair, I overlooked the `2 *` ([ref](https://github.com/lurk-lab/arecibo/blob/8e411d0e0c127397df4b85e3519df1c37d8b82ad/src/r1cs/mod.rs#L109-L115))